src/classlibnative/float/floatdouble.cpp

```diff
@@ -0,0 +1,300 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+// 
+// File: FloatDouble.cpp
+// 
+
+#include <common.h>
+
+#include "floatdouble.h"
+#include "floatclass_double.h"
+
+#define AS_BITFIELD(x)         (*((UINT64*)(&x)))
+
+#define IS_INFINITY(x)         ((AS_BITFIELD(x) & UI64(0x7FFFFFFFFFFFFFFF)) == UI64(0x7FF0000000000000))
+#define IS_ONE(x)              (AS_BITFIELD(x) == UI64(0x3FF0000000000000))
+#define IS_NEGATIVEONE(x)      (AS_BITFIELD(x) == UI64(0xBFF0000000000000))
+
+#ifdef _MSC_VER
+#pragma float_control(precise, off)
+#endif
+
+// Trigonometric Functions
+FCIMPL1_V(double, COMDouble::Acos, double x)
+    FCALL_CONTRACT;
+
+    return acos(x);
+FCIMPLEND
+
+FCIMPL1_V(double, COMDouble::Asin, double x)
+    FCALL_CONTRACT;
+
+    return asin(x);
+FCIMPLEND
+
+FCIMPL1_V(double, COMDouble::Atan, double x)
+    FCALL_CONTRACT;
+
+    return atan(x);
+FCIMPLEND
+
+FCIMPL2_VV(double, COMDouble::Atan2, double y, double x)
+    FCALL_CONTRACT;
+
+    // The cmath::atan2 method does not produce nan for atan2(+-inf,+-inf)
+    return (IS_INFINITY(y) && IS_INFINITY(x)) ? (y / x) : atan2(y, x);
```

Comment:
This code here, returning `nan` for `atan2(+-inf,+-inf)` goes against the recommended behavior in IEEE 754:2008 and what the MSVC compilers does.

A discussion should be brought up as to if this should maintain backwards compatibility or if this should follow the IEEE standard.

**Recommended Behavior (IEEE 754:2008 - Page 45)**:
* `atan2(±∞, −∞)` is `±3π/4`
* `atan2(±∞, +∞)` is `±π/4`